pubsub.go

```diff
@@ -326,7 +344,7 @@ func WithDiscovery(d discovery.Discovery, opts ...DiscoverOpt) Option {
 // WithEventTracer provides a tracer for the pubsub system
 func WithEventTracer(tracer EventTracer) Option {
 	return func(p *PubSub) error {
-		p.tracer = &pubsubTracer{tracer: tracer, pid: p.host.ID()}
+		p.tracer = &pubsubTracer{tracer: tracer, pid: p.host.ID(), msgID: p.msgID}
```

Comment:
one subtlety here is that the message ID option needs to be passed before the tracer option, otherwise the two will diverge.

Reply:
Good catch, see 6bd07a7 for your suggested fix.